### PR TITLE
IDEMPIERE-3336 - Tables missing EntityType column

### PR DESCRIPTION
**Pull request by muriloht**
adding entitytype field only for AD_StatusLineUsedIn AD_RelationType (script + model class)

### CODE REVIEW

File: migration/i7.1z/oracle/202009161600_IDEMPIERE-3336.sql [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+update AD_StatusLineUsedIn set EntityType='D' where AD_StatusLineUsedIn_ID = 200004;
+update AD_StatusLineUsedIn set EntityType='D' where AD_StatusLineUsedIn_ID = 200005;
+update AD_StatusLineUsedIn set EntityType='D' where AD_StatusLineUsedIn_ID = 200006;
+update AD_StatusLineUsedIn set EntityType='D' where AD_StatusLineUsedIn_ID = 200007;
+update AD_StatusLineUsedIn set EntityType='D' where AD_StatusLineUsedIn_ID = 200008;
+update AD_StatusLineUsedIn set EntityType='D' where AD_StatusLineUsedIn_ID = 200009;
+update AD_StatusLineUsedIn set EntityType='D' where AD_StatusLineUsedIn_ID = 200010;
+update AD_StatusLineUsedIn set EntityType='D' where AD_StatusLineUsedIn_ID = 200011;
+update AD_StatusLineUsedIn set EntityType='D' where AD_StatusLineUsedIn_ID = 200012;
+update AD_StatusLineUsedIn set EntityType='D' where AD_StatusLineUsedIn_ID = 1000000;
```
Comment (hengsin) [inline migration/i7.1z/oracle/202009161600_IDEMPIERE-3336.sql:122]: the usual convention should be EntityType='D' if _ID is < 1000000

File: migration/i7.1z/oracle/202009161600_IDEMPIERE-3336.sql [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+-- 08/04/2017 11h23min15s BRT
+INSERT INTO AD_Column_Trl (AD_Language,AD_Column_ID, Name, IsTranslated,AD_Client_ID,AD_Org_ID,Created,Createdby,Updated,UpdatedBy,AD_Column_Trl_UU ) SELECT l.AD_Language,t.AD_Column_ID, t.Name, 'N',t.AD_Client_ID,t.AD_Org_ID,t.Created,t.Createdby,t.Updated,t.UpdatedBy,Generate_UUID() FROM AD_Language l, AD_Column t WHERE l.IsActive='Y' AND l.IsSystemLanguage='Y' AND l.IsBaseLanguage='N' AND t.AD_Column_ID=212979 AND NOT EXISTS (SELECT * FROM AD_Column_Trl tt WHERE tt.AD_Language=l.AD_Language AND tt.AD_Column_ID=t.AD_Column_ID)
+;
+
+-- 08/04/2017 11h23min19s BRT
+UPDATE AD_Column SET FKConstraintName='EntityType_ADStatusLineUsedIn', FKConstraintType='N',Updated=TO_DATE('2017-04-08 11:23:19','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Column_ID=212979
+;
+
+-- 08/04/2017 11h23min19s BRT
+ALTER TABLE AD_StatusLineUsedIn ADD EntityType VARCHAR2(40) DEFAULT 'U'
```
Comment (globalqss) [inline migration/i7.1z/oracle/202009161600_IDEMPIERE-3336.sql:19]: For oracle it must use VARCHAR2(40 CHAR)

File: migration/i7.1z/oracle/202009161600_IDEMPIERE-3336.sql [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+-- 08/04/2017 11h26min12s BRT
+INSERT INTO AD_Column_Trl (AD_Language,AD_Column_ID, Name, IsTranslated,AD_Client_ID,AD_Org_ID,Created,Createdby,Updated,UpdatedBy,AD_Column_Trl_UU ) SELECT l.AD_Language,t.AD_Column_ID, t.Name, 'N',t.AD_Client_ID,t.AD_Org_ID,t.Created,t.Createdby,t.Updated,t.UpdatedBy,Generate_UUID() FROM AD_Language l, AD_Column t WHERE l.IsActive='Y' AND l.IsSystemLanguage='Y' AND l.IsBaseLanguage='N' AND t.AD_Column_ID=212980 AND NOT EXISTS (SELECT * FROM AD_Column_Trl tt WHERE tt.AD_Language=l.AD_Language AND tt.AD_Column_ID=t.AD_Column_ID)
+;
+
+-- 08/04/2017 11h26min16s BRT
+UPDATE AD_Column SET FKConstraintName='EntityType_ADRelationType', FKConstraintType='N',Updated=TO_DATE('2017-04-08 11:26:16','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Column_ID=212980
+;
+
+-- 08/04/2017 11h26min16s BRT
+ALTER TABLE AD_RelationType ADD EntityType VARCHAR2(40) DEFAULT 'U'
```
Comment (globalqss) [inline migration/i7.1z/oracle/202009161600_IDEMPIERE-3336.sql:79]: For oracle must be VARCHAR2(40 CHAR)

**Review by globalqss — CHANGES_REQUESTED** [review]

File: migration/i7.1z/oracle/202009161600_IDEMPIERE-3336.sql (hunk 1 above)
Comment (muriloht) [reply migration/i7.1z/oracle/202009161600_IDEMPIERE-3336.sql]: i'll take care in next script to use this convention. thank you for the review & tip!

File: migration/i7.1z/oracle/202009161600_IDEMPIERE-3336.sql (hunk 2 above)
Comment (muriloht) [reply migration/i7.1z/oracle/202009161600_IDEMPIERE-3336.sql]: :disappointed: I need to be more careful with Oracle scripts

File: migration/i7.1z/oracle/202009161600_IDEMPIERE-3336.sql (hunk 3 above)
Comment (muriloht) [reply migration/i7.1z/oracle/202009161600_IDEMPIERE-3336.sql]: :disappointed: I need to be more careful with Oracle scripts

**Comment by globalqss** [comment]
Hi @muriloht - the default value for entity type columns must be:
@SQL=select get_sysconfig('DEFAULT_ENTITYTYPE','U',0,0) from dual

Attached the corresponding patch to apply:
[mod258.txt](https://github.com/idempiere/idempiere/files/5251741/mod258.txt)

**Comment by muriloht** [comment]
> Hi @muriloht - the default value for entity type columns must be:
> @Sql=select get_sysconfig('DEFAULT_ENTITYTYPE','U',0,0) from dual
> 
> Attached the corresponding patch to apply:
> [mod258.txt](https://github.com/idempiere/idempiere/files/5251741/mod258.txt)

thanks @globalqss, I'm thinking about how to improove the page https://wiki.idempiere.org/en/Contributing_to_Trunk to have more information and help other people not make the same mistakes

**Review by globalqss — APPROVED** [review]

**Pull request merged**